src/lib/phdi-building-blocks/phdi_building_blocks/schemas.py

```diff
@@ -0,0 +1,251 @@
+        value = value
+
+    # Temporary hack to ensure no structured data is written using pyarrow.
+    # Currently Pyarrow does no support mixing non structure and structured data.
+    # https://github.com/awslabs/aws-data-wrangler/issues/463
+    # Will need to consider other methods of writing to parquet if this is and essential
+    # feature.
+    if type(value) == dict:
+        value = json.dumps(value)
+    elif type(value) == list:
+        value = ",".join(value)
+    return value
+
+
+def apply_schema_to_resource(resource: dict, schema: dict) -> dict:
+    """
+    Given a resource and a schema return a dictionary with values of the data
+    specified by the schema and associated keys defined by the variable name provided
+    by the schema.
+
+    :param resource: A FHIR resource on which to apply a schema.
+    :param schema: A schema specifying the desired values by FHIR resource type.
+    :return data: A dictionary containing the data extracted from the patient a long
+    with specified variable names.
+    """
+
+    data = {}
+    resource_schema = schema.get(resource.get("resourceType", ""))
+    if resource_schema is None:
+        return data
+    for field in resource_schema.keys():
+
+        path = resource_schema[field]["fhir_path"]
+        value = evaluate(resource, path)
+
+        if len(value) == 0:
+            data[resource_schema[field]["new_name"]] = ""
+        else:
+            selection_criteria = resource_schema[field]["selection_criteria"]
+            value = apply_selection_criteria(value, selection_criteria)
+            data[resource_schema[field]["new_name"]] = value
+
+    return data
+
+
+def make_resource_type_table(
+    resource_type: str,
+    schema: dict,
+    output_path: pathlib.Path,
+    output_format: Literal["parquet"],
+    credential_manager: AzureFhirserverCredentialManager,
+):
+    """
+    Given a FHIR resource type, schema, and FHIR server credential manager create a
+    table containing the field from resource type specified in the the schema.
+
+    :param resource_type: A FHIR resource type.
+    :param schema: A schema specifying the desired values by FHIR resource type.
+    :param credential_manager: A credential manager for
+    a FHIR server.
+    """
+
+    output_path.mkdir(parents=True, exist_ok=True)
+    output_file_name = output_path / f"{resource_type}.{output_format}"
+
+    query = f"/{resource_type}"
+    response = query_fhir_server(credential_manager, query)
+
+    additional_page = True
+    writer = None
+    while additional_page:
+        if response.status_code != 200:
+            log_fhir_server_error(response.status_code)
+            break
+
+        # Load queried data.
+        query_result = json.loads(response.content)
+        raw_schema_data = []
+
+        # Extract values specified by schema from each resource.
+        for resource in query_result["entry"]:
+            values_from_resource = apply_schema_to_resource(
```

Comment:
I think it would be nice to have a comment somewhere around here about the structure of `values_from_resource` and `raw_schema_data` - otherwise I need to jump around to figure that out.

Reply:
Agreed. Just added a comment about the form of `values_from_resource`.  I think given that we initialize `raw_schema_data`, now simply `data`, as a list that we explicitly append `values_from_resource` it's form is clear. Do you agree, or would you like to see additional documentation?